### ISSUE TITLE
Deprecated components (Will not be maintained actively)

### ISSUE THREAD

**Issue by SeanNaren** [PyTorchLightning/lightning-bolts#719]
Related to #712 

Below I've gone through all the different modules available, and come up with a list of deprecated modules. I've been lenient in my choices, primarily to allow some discussions if any should be moved around. **If you think the list is wrong, please comment!**

Deprecated modules will be made clear within a new documentation page containing a list of all deprecated modules (+ links eventually). This hopefully allows us to draw a line between model specific implementations + applicable research components within a field.

Also important is having a gateway for users to where they can get such implementations; i.e Flash. Any suggestions for RL based implementations? There must be a nice open source repo for Lightning + RL.

### Callbacks

- BatchGradientVerification
- ConfusedLogitCallback
- ModuleDataMonitor
- TrainingDataMonitor
- PrintTableMetricsCallback

### Datamodules

- BinaryEMNISTDataModule
- BinaryMNISTDataModule
- CIFAR10DataModule
- CityscapesDataModule
- EMNISTDataModule
- FashionMNISTDataModule
- ImagenetDataModule
- KittiDataModule
- MNISTDataModule
- SklearnDataModule
- SSLImagenetDataModule
- STL10DataModule
- VisionDataModule
- VOCDetectionDataModule

### Datasets

- LightDataset
- CIFAR10
- ConcatDataset
- DummyDataset
- BinaryEMNIST
- UnlabeledImagenet
- KittiDataset
- BinaryMNIST
- CIFAR10Mixed

### Optimizers

- LinearWarmupCosineAnnealingLR
- linear_warmup_decay

### Callbacks

- BYOLMAWeightUpdate
- SSLOnlineEvaluator
- LatentDimInterpolator	
- TensorboardGenerativeModelImageSampler
- KNNOnlineEvaluator

### Datamodules

- AsynchronousLoader
- ExperienceSourceDataset
- ExperienceSource
- DiscountedExperienceSource

### Losses

- iou_loss
- giou_loss
- dqn_loss
- double_dqn_loss
- per_dqn_loss
- nt_xent_loss
- CPCTask
- AmdimNCELoss
- FeatureMapContrastiveTask

### Metrics

- mean (use [torchmetrics](https://github.com/PyTorchLightning/metrics))
- accuracy (use [torchmetrics](https://github.com/PyTorchLightning/metrics))
- precision_at_k (use [torchmetrics](https://github.com/PyTorchLightning/metrics))
- iou (use [torchmetrics](https://github.com/PyTorchLightning/metrics))
- giou (use [torchmetrics](https://github.com/PyTorchLightning/metrics))

### Models

- AE
- VAE
- FasterRCNN
- GAN
- DCGAN
- Pix2Pix
- LinearRegression
- LogisticRegression
- AdvantageActorCritic
- DoubleDQN
- DQN
- DuelingDQN
- NoisyDQN
- PERDQN
- PPO
- Reinforce
- VanillaPolicyGradient
- AMDIM
- BYOL
- CPC_v2
- Moco_v2
- SimCLR
- SimSiam
- SwAV
- SSLEvaluator
- ImageGPT
- PixelCNN
- SemSegment
- UNet
- LitMNIST

### Optimizers

- LARS

 ### Transforms

- RandomTranslateWithReflect
- Patchify

**Comment by SeanNaren** [comment]
cc @Borda @akihironitta @tchaton @PyTorchLightning/core-bolts
